scripts/auth0connections/auth0connections.py

```diff
@@ -29,26 +30,44 @@ def get_client():
 
 
 def render_local_connections(config):
+    '''Returns full connection dicts, from the connections listed in the config
+    file, and rendered using the template specified.
+    
+    Uses a template specified in the connection config
+    e.g. 'template_name: hmpps-auth' in the connection configuration
+         means use the template in: ./connection_templates/hmmps-auth/
+    '''
     connections = {}
-    connections_root = Path(__file__).cwd() / Path("connections")
-    connection_dirs = [entry for entry in connections_root.iterdir() if entry.is_dir()]
-
-    for connection in connection_dirs:
-        connection_name = connection.stem
-        scripts = connection.glob("*.js")
+    template_root = Path(__file__).cwd() / Path("connection_templates")
+    template_dirs = dict((entry.stem, entry) for entry in template_root.iterdir() if entry.is_dir())
+
+    for connection_name in config:
+        connection = config[connection_name]
+        connection['name'] = connection_name
+        template_name = connection['connection_template']
+        try:
+            template_path = template_dirs[template_name]
+        except KeyError:
+            print(f'template_name: "{template_name}" is specified in the config, but no such template exists in {template_root}')
+            sys.exit(1)
```

Comment:
again potentially you may change to not use `try`/`except` and simply check if the key is in the dictionary which looks less drastic. Something like:

```python
if template_name not in template_dirs:
    print([...])
    sys.exit(1)

```